### PR TITLE
Revised stubs for geoip2 third party library

### PR DESCRIPTION
**Pull request by jolaf**
Followup to #3299.

Approved by the developer, see https://github.com/maxmind/GeoIP2-python/issues/70

Also in this PR:

- Stubs for `maxminnddb` moved from `third_party/3` to `third_party/2and3` with appropriate adjustement of `str` replaced with `Text`.

- `stdlib/3/ipaddress.pyi` got copied to `third_party/2` as this backport is necessary for `maxminddb` on Python 2. Both copies are adjusted by `str` replaced with `Text`.

- Both `maxminddb` and `geoip2` stubs got rid of extra comments.

### CODE REVIEW

**Comment by jolaf** [comment]
Nope, it doesn't work with Python2:
```
third_party/2and3/geoip2/mixins.pyi:6: error: Dynamic metaclass not supported for 'SimpleEquality'

third_party/2and3/geoip2/records.pyi:11: error: Dynamic metaclass not supported for 'Record'

third_party/2and3/geoip2/records.pyi:16: error: Dynamic metaclass not supported for 'PlaceRecord'

third_party/2and3/geoip2/models.pyi:34: error: Dynamic metaclass not supported for 'SimpleModel'

third_party/2and3/geoip2/database.pyi:6: error: Cannot find module named 'maxminddb.reader'
```
I suggest moving the stubs back to `third_party/3`.

**Comment by JelleZijlstra** [comment]
You can probably just remove the `__metaclass__` stuff; I don't think these metaclasses really make a difference in the stubs.

**Comment by jolaf** [comment]
> You can probably just remove the `__metaclass__` stuff; I don't think these metaclasses really make a difference in the stubs.
Well, to make this work, I had to also move `maxmindb` from `third_party/3` to `third_party/2and3`, and also copy `ipaddress` (that `maxminddb` depends on) from `stdlib/3` to `third_party/2`, as it is a backport.

**Comment by jolaf** [comment]
Can we please move on with this?
It would be very nice if this goes into the upcoming `mypy` release.
Thanks!

File: third_party/2/ipaddress.pyi [hunk 1]
```diff
@@ -0,0 +1,149 @@
+from typing import (Any, Container, Generic, Iterable, Iterator, Optional,
```
Comment (JelleZijlstra) [inline third_party/2/ipaddress.pyi]: Could you add this file to check_consistent.py to make sure it stays in sync with third_party/3/ipaddress.pyi?

File: third_party/2and3/maxminddb/__init__.pyi [hunk 2]
```diff
@@ -0,0 +1,9 @@
+# Stubs for maxminddb (Python 2 and 3)
```
Comment (JelleZijlstra) [inline third_party/2and3/maxminddb/__init__.pyi]: Let's just remove these comments; they're not useful.

File: third_party/2/ipaddress.pyi (hunk 1 above)
Comment (jolaf) [reply third_party/2/ipaddress.pyi]: Done.

File: third_party/2and3/maxminddb/__init__.pyi (hunk 2 above)
Comment (jolaf) [reply third_party/2and3/maxminddb/__init__.pyi]: Done.

**Comment by jolaf** [comment]
> Sure, just a few small comments
All fixed.

Thank you!

**Pull request merged**